ArduCopter/Attitude.cpp

```diff
@@ -128,10 +128,36 @@ void Copter::update_throttle_hover()
     if (throttle > 0.0f && abs(climb_rate) < 60 && labs(ahrs.roll_sensor) < 500 && labs(ahrs.pitch_sensor) < 500) {
         // Can we set the time constant automatically
         motors->update_throttle_hover(0.01f);
+        
+        // Learn the wattage required to hover based on current and estimated resting voltage
+        if (g2.hover_watt_learn != HOVER_WATT_LEARN_DISABLED) {
+            g2.hover_watt = calc_hover_watts();
+        }
+
     }
 #endif
 }
 
+float Copter::calc_hover_watts()
+{
+    // init the variables
+    static uint16_t loop_count = 0;
+    static uint16_t watts_avg = 0;
+    static uint16_t watts_running_total = 0;
+    
+    // Running average watts calculation
+    watts_running_total += battery.watts();
+    loop_count ++;
+    watts_avg = watts_running_total / loop_count;
```

Comment:
divide by 0 on first call

Reply:
well, loop_count is incremented but this i snot style conform because of the space